### PR TITLE
Astro 1947 form pattern

### PR DESCRIPTION
**Pull request by markacianfrani**
## Brief Description

Adds form pattern examples for Vue and vanilla js

## JIRA Link

[ASTRO-1947](https://rocketcom.atlassian.net/browse/ASTRO-1947)

## Motivation and Context

Shows concrete examples of data binding for every form and how to setup validation. 

## Issues and Limitations

Not sure if these are better suited as part of the starter kits or not. We can definitely reevaluate. There's some maintenance burden with using Codesandbox I think

## Types of changes

-   [ ] Bug fix
-   [x] New feature
-   [ ] Breaking change

## Checklist

-   [x] This PR adds or removes a Storybook story.
-   [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by micahjones13** [comment]
In the preview, submitting the HTML form results in a 405 error, but the alert for vue works.

**Comment by markacianfrani** [comment]
> In the preview, submitting the HTML form results in a 405 error, but the alert for vue works.

I didn't want to add a bunch of boilerplate JS just for catching the form event and outputting it so instead it just submits to / which doesn't exist. In the real world, you'd have your own api endpoint there. I do like the idea of it being consistent with the vue example, but I dont think we should add js just for that. I could see someone inexperienced copy/pasting this and wondering why its not submitting. What do you think?

**Comment by micahjones13** [comment]
> > In the preview, submitting the HTML form results in a 405 error, but the alert for vue works.
> 
> I didn't want to add a bunch of boilerplate JS just for catching the form event and outputting it so instead it just submits to / which doesn't exist. In the real world, you'd have your own api endpoint there. I do like the idea of it being consistent with the vue example, but I dont think we should add js just for that. I could see someone inexperienced copy/pasting this and wondering why its not submitting. What do you think?

I think it's fine without a real submission. If we're worried about someone copy/pasting it we could make a not of it in docs.

File: src/stories/patterns/forms/form-svelte.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  </rux-tab-panel>
+  
+  <rux-tab-panel aria-labelledby="tab-id-2">
+
+## Example
+
+An example using [Svelte](https://svelte.dev/) + [Yup](https://github.com/jquense/yup) for easy validation.
+
+<iframe src="https://codesandbox.io/s/astro-form-in-svelte-with-validation-wvwyd?file=/Form.svelte"
```
Comment (micahjones13) [inline src/stories/patterns/forms/form-svelte.stories.mdx:72]: @NicDominguez This has all of the form elements but only the input fields are being validated. I'm not sure we need to include elements in the form validation example that aren't actually being validated. The vue validation example only has rux-inputs included.

Comment (NicDominguez) [reply src/stories/patterns/forms/form-svelte.stories.mdx]: @micahjones13 This morning @markacianfrani forked that validation sandbox and added a couple features while removing the non-validated components. It looks like he hadn't updated the url to the new sandbox yet so I just pushed the change. The non-input components were going to be a bit more work to figure out validation for that we don't want to do at this point.

**Review by micahjones13 — APPROVED** [review]

**Pull request merged**